visualization/app/codeCharta/core/tooltip/tooltipService.js

```diff
@@ -11,27 +10,61 @@ class TooltipService {
     /**
      * @constructor
      */
-    constructor(){
-
-        /**
-         * All available tooltips.
-         * @type {{Statements: Tooltip, RLOC: Tooltip, MCC: Tooltip}}
-         */
-        this.tooltips = {
-            Statements: new Tooltip("Statements", "Number of Statements"),
-            RLOC: new Tooltip("RLOC", "Real Lines of Code"),
-            MCC: new Tooltip("MCC", "MacCabe Complexity or cyclomatic complexity"),
-        };
+    constructor($rootScope, urlService){
+
+        this.tooltips = [];
+        var ctx = this;
+
+        urlService.getFileDataFromFile("tooltips.json").then(
+
+            //resolve
+            (data) => {
+                ctx.tooltips = data;
+                $rootScope.$broadcast("tooltips-changed", this.tooltips);
+            },
+            //reject
+            () => {
+                window.alert("error loading tooltips.json");
+            }
+
+        );
 
     }
 
     /**
      * returns the tooltip description related to the given key
      * @param {String} key
-     * @returns {string} description
+     * @returns {string} the description related to the given key
      */
     getTooltipTextByKey(key) {
-        return this.tooltips[key] ? this.tooltips[key].getTooltip() : "no description";
+
+        var patt = new RegExp(/_\S*_/);
+
+        if(this.tooltips[key]){
+
+            var res = this.tooltips[key];
+
+
+            while (patt.test(res)) {
+                res = res.replace(/_(.*?)_/, this.replaceString.bind(this));
+            }
+            return res;
+
+        } else {
+            return "no description";
+        }
+
+    }
+
+    /**
+     * Function used for recursiveness with getTooltipTexByKey
+     * @param a
+     * @param {String} b given key
+     * @returns {string}
+     */
+    replaceString(a, b) {
+        var rep = this.getTooltipTextByKey(b);
+        return rep;
     }
```

Comment:
You can shorten this function by using just a single line:
`return this.getTooltipTextByKey(b);`